spec/lib/ordering_helper_spec.rb

```diff
@@ -0,0 +1,159 @@
+    end
+    it 'pivots down and leaves neutral elements' do
+      eq_model1 = FactoryGirl.build_stubbed(:equipment_model,
+                                            ordering: 1)
+      eq_model2 = FactoryGirl.build_stubbed(:equipment_model,
+                                            ordering: 3)
+      eq_model3 = FactoryGirl.build_stubbed(:equipment_model,
+                                            ordering: 5)
+
+      allow(eq_model1).to receive(:update_attribute)
+      allow(eq_model2).to receive(:update_attribute)
+      allow(eq_model3).to receive(:update_attribute)
+
+      helper = OrderingHelper.new(eq_model1)
+      allow(helper).to receive(:ordering).and_return(1)
+      allow(helper).to receive(:successor).and_return(eq_model2)
+      allow(helper).to receive(:rel_successor).and_return(3)
+      allow(helper).to receive(:cat_last).and_return(5)
+
+      helper.down
+      expect(eq_model1).to have_received(:update_attribute).with('ordering', 3)
+      expect(eq_model2).to have_received(:update_attribute).with('ordering', 1)
+      expect(eq_model3).not_to have_received(:update_attribute)
+    end
+  end
+  describe 'deactivate' do
+    it 'identifies successors' do
+      category = FactoryGirl.create(:category)
+      FactoryGirl.create(:equipment_model,
+                         category: category,
+                         ordering: 1)
+      eq_model2 = FactoryGirl.create(:equipment_model,
+                                     category: category,
+                                     ordering: 3)
+      eq_model3 = FactoryGirl.create(:equipment_model,
+                                     category: category,
+                                     ordering: 5)
+      expect(OrderingHelper.new(eq_model2).successors).to eq([eq_model3])
+    end
+    it 'handles ordering on deactivation' do
+      eq_model1 = FactoryGirl.build_stubbed(:equipment_model,
+                                            ordering: 1)
+      eq_model2 = FactoryGirl.build_stubbed(:equipment_model,
+                                            ordering: 3)
+      eq_model3 = FactoryGirl.build_stubbed(:equipment_model,
+                                            ordering: 5)
+      allow(eq_model1).to receive(:update_attribute)
+      allow(eq_model2).to receive(:update_attribute)
+      allow(eq_model2).to receive(:deactivate)
+      allow(eq_model3).to receive(:update_attribute)
+      helper = OrderingHelper.new(eq_model2)
+      allow(helper).to receive(:cat_last).and_return(5)
+      allow(helper).to receive(:successors).and_return([eq_model3])
+      allow(helper).to receive(:rel_predecessor).with(5).and_return(3)
+      helper.deactivate_order
+      expect(eq_model1).not_to have_received(:update_attribute)
+      expect(eq_model2).to have_received(:update_attribute).with('ordering', 5)
+      expect(eq_model3).to have_received(:update_attribute).with('ordering', 3)
+    end
+  end
+end
```

Comment:
these tests are very good and thorough, but they're quite long -- can you try to break them up a bit, either into smaller tests or by pulling out helper methods?